spec/controllers/announcements_controller_spec.rb

```diff
@@ -129,5 +149,8 @@
       end
       it_behaves_like 'access denied'
     end
+    context 'GET hide' do
```

Comment:
Note that this test only checks for patrons (see line 123 above), not all user roles. That's why it didn't catch the others. We should probably write tests for all roles... as this is a common problem I'm going to open a new issue to write an RSpec helper to deal with it.